hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoGetAccountInfoHandler.java

```diff
@@ -58,13 +106,289 @@ public Response createEmptyResponse(@NonNull final ResponseHeader header) {
+            info.receiverSigRequired(account.receiverSigRequired());
+            info.deleted(account.deleted());
+            info.memo(account.memo());
+            info.autoRenewPeriod(Duration.newBuilder().seconds(account.autoRenewSecs()));
+            info.balance(account.tinybarBalance());
+            info.expirationTime(Timestamp.newBuilder().seconds(account.expiry()));
+            info.contractAccountID(getContractAccountId(account, account.alias()));
+            info.ownedNfts(account.numberOwnedNfts());
+            info.maxAutomaticTokenAssociations(account.maxAutoAssociations());
+            info.ethereumNonce(account.ethereumNonce());
+            //  info.proxyAccountID(); Deprecated
+            info.alias(account.alias());
+            info.tokenRelationships(getTokenRelationship(tokensConfig, account, tokenStore, tokenRelationStore));
+            info.stakingInfo(getStakingInfo(account, rewardCalculator, readableStakingInfoStore));
+            return Optional.of(info.build());
+        }
+    }
+
+    /**
+     * get TokenRelationship of an Account
+     * @param tokenConfig use TokenConfig to get maxRelsPerInfoQuery value
+     * @param account the account to be calculated from
+     * @param readableTokenStore readable token store
+     * @param tokenRelationStore token relation store
+     * @return ArrayList of TokenRelationship object
+     */
+    private List<TokenRelationship> getTokenRelationship(
+            @NonNull final TokensConfig tokenConfig,
+            @NonNull final Account account,
+            @NonNull final ReadableTokenStore readableTokenStore,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore) {
+        requireNonNull(account);
+        requireNonNull(tokenConfig);
+        requireNonNull(readableTokenStore);
+        requireNonNull(tokenRelationStore);
+        final var ret = new ArrayList<TokenRelationship>();
+        var tokenNum = account.headTokenNumber();
+        int count = 0;
+        Optional<TokenRelation> tokenRelation;
+        Token token; // token from readableToken store by tokenID
+        TokenID tokenID; // build from tokenNum
+        AccountID accountID; // build from accountNumber
+        while (tokenNum != 0 && count < tokenConfig.maxRelsPerInfoQuery()) {
+            accountID =
+                    AccountID.newBuilder().accountNum(account.accountNumber()).build();
+            tokenID = TokenID.newBuilder().tokenNum(tokenNum).build();
+            tokenRelation = tokenRelationStore.get(accountID, tokenID);
+            if (tokenRelation.isPresent()) {
+                token = readableTokenStore.get(tokenID);
+                if (token != null) {
+                    addTokenRelation(ret, token, tokenRelation.get(), tokenNum);
+                }
+                tokenNum = tokenRelation.get().nextToken();
+            } else {
+                break;
+            }
+            count++;
+        }
+        return ret;
```

Comment:
seems like iterating token relations is common for both accountBalance and account info. It can be put in one common place. Not blocker for this PR

Reply:
Will make the change in TokenGetInfoHandler pr.